### DISCUSSION TITLE
CouplingTermObstacleAvoidance2D is not working

### DISCUSSION THREAD

**Discussion by perfectism13** [dfki-ric/movement_primitives#65] [Q&A]
I used CouplingTermObstacleAvoidance2D  as follows:![image](https://user-images.githubusercontent.com/31556137/173241137-16ace4f5-cf3e-4f2c-bb32-3a59edaa9e40.png),but it does not working.
![image](https://user-images.githubusercontent.com/31556137/173241173-d276218a-aa4b-49bd-8352-7fb544fd0474.png)
When I change the Y to a simple sint., it working.
![image](https://user-images.githubusercontent.com/31556137/173241241-4c68f978-2754-494b-bf29-7cd361931cac.png)

**Comment by AlexanderFabisch** [comment]
Hi.

> ,but it does not working.

This can happen. Try to change the parameters gamma and beta of the coupling terms. The default values come from the paper "Dynamical movement primitives: learning attractor models for motor behaviors" of Ijspeert et al.

>  When I change the Y to a simple sint., it working.

What is sint?

**Comment by perfectism13** [comment]
Thanks, I will have a try.  Sint means I used a simple Y as follows.
`dt = 0.01
execution_time = 1.0
T = np.arange(0, execution_time + dt, dt)
Y = np.column_stack((np.sin(np.pi * T), -np.cos(np.pi * T)))`
